src/App.tsx

```diff
@@ -1,39 +1,57 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
+interface State {
+  clockName: string;
+  hasClock: boolean;
+}
+
+export class App extends Component<{}, State> {
+  state: Readonly<State> = {
+    clockName: getRandomName(),
+    hasClock: true,
+  };
+
+  timerId = 0;
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
+  componentDidMount() {
+    this.timerId = this.setTimer();
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+    document.addEventListener('contextmenu', this.contextEvent);
+    document.addEventListener('click', this.clickEvent);
+  }
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  setTimer = () => {
+    return window.setInterval(() => {
+      this.setState({ clockName: getRandomName() });
+    }, 3300);
+  };
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  clickEvent = () => {
```

Comment:
Name of the method should begin with a verb
```suggestion
  handleClickEvent = () => {
```